packages/mui-base/src/utils/useAnchorPositioning.ts

```diff
@@ -179,14 +181,14 @@ export function useAnchorPositioning(
         const transformX = arrowX + arrowWidth / 2;
         const transformY = arrowY + arrowHeight;
 
-        const transformOrigin = {
+        const arrowOrigin = {
           top: `${transformX}px calc(100% + ${arrowHeight}px)`,
           bottom: `${transformX}px ${-arrowHeight}px`,
           left: `calc(100% + ${arrowHeight}px) ${transformY}px`,
           right: `${-arrowHeight}px ${transformY}px`,
         }[currentRenderedSide];
 
-        elements.floating.style.setProperty('--transform-origin', transformOrigin);
+        elements.floating.style.setProperty(`--${name}-arrow-origin`, arrowOrigin);
 
```

Comment:
The arrow origin and transform origin are a bit different. For edge-aligned popups, you want the `transform-origin` to be at the edge, not arrow's point/anchor center.